interapp-frontend/src/app/page.tsx

```diff
@@ -2,24 +2,52 @@
 import { useContext } from 'react';
 import { AuthContext } from '@providers/AuthProvider/AuthProvider';
 import { useRouter } from 'next/navigation';
-import UnderConstruction from '@components/UnderConstruction/UnderContruction';
+
+import AnnouncementList from '@components/AnnouncementList/AnnouncementList'
+import ServiceList from '@components/ServiceList/ServiceList'
+import AttendanceList from '@components/AttendanceList/AttendanceList'
+
+import { Text, Box } from '@mantine/core';
+import Image from 'next/image';
+import './styles.css';
 
 export default function Home() {
   const router = useRouter();
   const { user, logout } = useContext(AuthContext);
-  return (
-    <>
-      <UnderConstruction />
-      {user && (
-        <>
-          <h1>{user.username}</h1>
-          <h2>{user.user_id}</h2>
-          <h2>{user.email}</h2>
-          <h2>{user.permissions}</h2>
-          <h3>{user.verified}</h3>
-          <h3>{user.service_hours}</h3>
-        </>
-      )}
-    </>
-  );
+
+  // checks whether user is an interact member or a visitor; true if interact member, false if visitor
+  const has_permission = user?.permissions.some(element => {
```

Comment:
1. consider checking if user is truthy (ie. not null)
`` !!user && user.permissions.some(...) ``
because the current syntax (``user?.permissions. ...``) can return undefined instead of a boolean value
2. consider memoising this value to persist across renders until user has changed. (not necessary but i would recommend)